### PR TITLE
[IOTDB-3809] Make query error more specific in Coordinator

### PR DESCRIPTION
**Pull request by JackieTien97**
In this pr, we print more specific error message in coordinator instead of fragment instance cannot be dispatched.
You can more details about it in [feishu doc](https://apache-iotdb.feishu.cn/docx/doxcnAWBTm9MYVKuSNUPmglCNUh).

### CODE REVIEW

File: server/src/main/java/org/apache/iotdb/db/mpp/execution/exchange/SharedTsBlockQueue.java [hunk 1]
```diff
@@ -152,12 +152,12 @@ public ListenableFuture<Void> add(TsBlock tsBlock) {
     return blockedOnMemory;
   }
 
-  /** Destroy the queue and cancel the future. */
-  public void destroy() {
-    if (destroyed) {
+  /** Destroy the queue and complete the future. Should only be called in normal case */
```
Comment (xingtanzjr) [inline server/src/main/java/org/apache/iotdb/db/mpp/execution/exchange/SharedTsBlockQueue.java:155]: `abnormal case`

File: server/src/main/java/org/apache/iotdb/db/mpp/execution/exchange/SharedTsBlockQueue.java [hunk 2]
```diff
@@ -172,4 +172,27 @@ public void destroy() {
       bufferRetainedSizeInBytes = 0;
     }
   }
+
+  // TODO add Throwable t as a parameter of this method, and then call blocked.setException(t);
+  // instead of blocked.cancel(true);
+  /** Destroy the queue and cancel the future. Should only be called in normal case */
```
Comment (xingtanzjr) [inline server/src/main/java/org/apache/iotdb/db/mpp/execution/exchange/SharedTsBlockQueue.java:178]: `abnormal case`

File: server/src/main/java/org/apache/iotdb/db/mpp/plan/execution/QueryExecution.java [hunk 3]
```diff
@@ -280,18 +281,30 @@ private void releaseResource() {
...
-  public Optional<TsBlock> getBatchResult() {
+  public Optional<TsBlock> getBatchResult() throws IoTDBException {
+    checkArgument(resultHandle != null, "ResultHandle in Coordinator should be init firstly.");
     // iterate until we get a non-nullable TsBlock or result is finished
     while (true) {
       try {
-        if (resultHandle == null || resultHandle.isAborted() || resultHandle.isFinished()) {
+        if (resultHandle.isAborted()) {
+          logger.info("resultHandle for client is aborted");
+          stateMachine.transitionToAborted();
```
Comment (xingtanzjr) [inline server/src/main/java/org/apache/iotdb/db/mpp/plan/execution/QueryExecution.java:291]: Does that the resultHandle is `aborted` mean the stateMachine is `Failed` ? 

I mean, the transitionToAborted won't take effect ?

File: server/src/main/java/org/apache/iotdb/db/service/thrift/impl/DataNodeInternalRPCServiceImpl.java [hunk 4]
```diff
@@ -208,11 +246,13 @@ public TSendPlanNodeResp sendPlanNode(TSendPlanNodeReq req) {
   @Override
   public TFragmentInstanceStateResp fetchFragmentInstanceState(TFetchFragmentInstanceStateReq req) {
     FragmentInstanceId instanceId = FragmentInstanceId.fromThrift(req.fragmentInstanceId);
-    try (SetThreadName threadName = new SetThreadName(instanceId.getFullId())) {
```
Comment (xingtanzjr) [inline server/src/main/java/org/apache/iotdb/db/service/thrift/impl/DataNodeInternalRPCServiceImpl.java:211]: Why do we remove this line ?

**Review by xingtanzjr — APPROVED** [review]

File: server/src/main/java/org/apache/iotdb/db/mpp/plan/execution/QueryExecution.java (hunk 3 above)
Comment (JackieTien97) [reply server/src/main/java/org/apache/iotdb/db/mpp/plan/execution/QueryExecution.java:291]: Yeap, I also think so.

File: server/src/main/java/org/apache/iotdb/db/service/thrift/impl/DataNodeInternalRPCServiceImpl.java (hunk 4 above)
Comment (JackieTien97) [reply server/src/main/java/org/apache/iotdb/db/service/thrift/impl/DataNodeInternalRPCServiceImpl.java:211]: It seems that there is no log printed in this method.

File: server/src/main/java/org/apache/iotdb/db/mpp/execution/exchange/SharedTsBlockQueue.java (hunk 2 above)
Comment (JackieTien97) [reply server/src/main/java/org/apache/iotdb/db/mpp/execution/exchange/SharedTsBlockQueue.java]: fixed

**Pull request merged**